src/main/java/org/cryptomator/ui/recoverykey/RecoveryKeyIsHubVaultController.java

```diff
@@ -0,0 +1,46 @@
+package org.cryptomator.ui.recoverykey;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javafx.beans.property.ObjectProperty;
+import javafx.fxml.FXML;
+import javafx.scene.Scene;
+import javafx.stage.Stage;
+import java.util.ResourceBundle;
+
+import dagger.Lazy;
+import org.cryptomator.common.RecoverUtil;
+import org.cryptomator.ui.common.FxController;
+import org.cryptomator.ui.common.FxmlFile;
+import org.cryptomator.ui.common.FxmlScene;
+
+@RecoveryKeyScoped
+public class RecoveryKeyIsHubVaultController implements FxController {
```

Comment:
The name suggests that the window is displayed, if a hub vault is detected. I assume this is shown _before_ it is checked (because otherwise we have a simple dialog).